src/backend/marsha/core/views.py

```diff
@@ -175,132 +298,121 @@ def _get_base_app_data(self):
 
         return app_data
 
-    def _get_prepare_to_render(self, app_data):
-        """Utility to build expected params to render_to_response"""
-        return {
-            "app_data": json.dumps(app_data),
-            "static_base_url": f"{settings.STATIC_URL}js/build/",
-            "external_javascript_scripts": settings.EXTERNAL_JAVASCRIPT_SCRIPTS,
-        }
+    def _get_resource_data(self, resource, session_id, is_admin=False, user_id=None):
+        """Serializes the resource, provided some context, for the frontend app data."""
+        if not resource:
+            return None
+
+        return self.serializer_class(
+            resource,
+            context={
+                "is_admin": is_admin,
+                "user_id": user_id,
+                "session_id": session_id,
+            },
+        ).data
 
-    def get_context_data(self):
-        """Build context for template rendering of configuration data for the frontend.
 
-        Returns
-        -------
-        dictionary
-            context with configuration data for the frontend
+@method_decorator(csrf_exempt, name="dispatch")
+@method_decorator(xframe_options_exempt, name="dispatch")
+class BaseLTIView(BaseModelResourceView, ABC):
+    """Base LTI view called to serve a resource on an LTI context
+    and how to use it by the front application.
 
-        """
+    This view answer to POST request only.
 
-        def _manage_exception(error):
-            logger.warning(str(error))
-            app_data = self._get_base_app_data()
-            app_data.update(
-                {
-                    "resource": None,
-                    "state": "error",
-                }
-            )
-            return app_data
+    It is designed to work as a React single page application.
+    """
+
+    @property
+    def cache_key(self):
+        """Cache key from view context."""
+
+        return (
+            "app_data"
+            f"|{self.model.__name__}"
+            f"|{self.lti.get_consumer_site().domain}"
+            f"|{self.lti.context_id}"
+            f"|{self.lti.resource_id}"
+        )
```

Comment:
Maybe we could use in BaseModelResourceView something like 
```python
    @staticmethod
    def build_cache_key(*keys):
        return "|".join(str(key) for key in keys)
```
So we can do
```suggestion
        return self.build_cache_key(
            "app_data",
            self.model.__name__,
            self.lti.get_consumer_site().domain,
            self.lti.context_id,
            self.lti.resource_id,
        )
```

Reply:
Thanks, I once tried to use the `join` to create cache key, but it was not helping readability. Using a `build_cache_key` makes it way better :+1: